background/hardware.tex

```diff
@@ -136,25 +136,34 @@ \subsection{Evaluation boards}
+The Cortex-M3 and its memory can be found in the upper left corner, it can be seen from the diagram that the \gls{mcu} is connected to all other peripherals over a 32-bit bus.
+This section will shortly introduce the peripherals that are relevant in this project, we will describe what they can be used for and why they are important, but we will not go into detail of how they work.
+
+\begin{figure}[H]
+\begin{center}
+\includegraphics[width=0.8\textwidth]{figures/gg_block_diagram}
+\end{center}
+\caption{Giant Gecko block diagram \cite{Labs}.}
 \label{fig:efm32gg990_block_diagram}
 \end{figure}
 
-For this project, the porting of library bindings are only interesting if the peripherals exposed by
-the library are actually going to be used. The two boards we have access to implements many external
-sensors and peripherals that are all controlled through \emlib. It is only interesting to write
-bindings and abstractions for the core functionality of \emlib and the parts of the library that
-lets us control peripherals that we use in throughout the projects.
+Communication is important in every modern computing system, the {\gecko} supports a handful of different serial communication protocols through different peripherals.
+The \gls{usart}, and the \gls{uart}, together with the \gls{i2c}, are serial protocols that allows for efficient communication between a high range of external devices.
+Different areas of use can be e.g. reading and writing to SD memory cards or transferring data over a serial interface to a computer.
+Peripherals like the \gls{usart} and the \gls{uart} can be used in combination with \gls{dma}, which is useful for energy constraint applications.
+This combination make it possible to implement applications that can gather data from a range of external sensors, and transfer or store it to other peripherals with minimal interaction from the \gls{cpu}.
+
+\gls{gpio} is used for pin configuration and manipulation of the \gls{mcu} pins, and are often used in order to configure the pins that are required by other peripherals.
+Simple Push Buttons and LEDs are configured with the \gls{gpio}, and e.g. the \gls{usart} needs to have configured at least two \gls{gpio} pins for both receiving and transmitting data.
+
+Many applications are used to gather data by measuring different sensors and signals.
+The \gls{adc} is a peripheral that is used to measure analogue signals, like e.g. sound or light intensity, and convert them to digital signals.
+Oppositely, the \gls{dac} can be used to convert digital signals back to e.g. analogue sound signals.
+
+Timers are another important peripheral type, the {\gecko} supports a number different timers that operate with different frequencies and at different energy levels.
+The Timer peripheral is used to keep track of timing and counting of different events, without intervention of the \gls{cpu}, it can e.g. be used to initiate \gls{dma} transfer over \gls{uart} at timed intervals.
```

Comment:
What about: 

`The Timer peripheral is used to generate signals at specified frequencies. These can be software interrupts to the CPU or signals to peripherals which in turn, without intervention of the \gls{cpu}, it can be used to e.g. initiate \gls{dma} transfer over \gls{uart} at timely intervals.`